### PR TITLE
Warn of payments to exchanges without memo #48

### PR DESCRIPTION
**Pull request by ghost**
Just the minimum to prevent users making the mistake of sending a payment to one of the major exchanges while forgetting to fill out the memo field.

### CODE REVIEW

**Review by bartekn — COMMENTED** [review]
Looks good but I think we shouldn't enforce it. It's possible that some exchanges are using Account Viewer with Ledger to transfer money between accounts and we should allow them to do it.

So if destination is an exchange we should display a warning but still allow sending transaction without memo. What do you think?

**Comment by ghost** [comment]
Yes, I considered that too. I've changed the PR now. When a known account is entered the memo field is shown automatically along with a warning message that the destination requires a memo. Users can still proceed without memo by clicking cancel button above the memo field.

File: controllers/send-widget.controller.es6 [hunk 1]
```diff
@@ -123,8 +124,18 @@ export default class SendWidgetController {
...
+        if (this.destination in knownAccounts && knownAccounts[this.destination].requiredMemoType) {
+          this.memo = true;
+          this.memoType = knownAccounts[this.destination].requiredMemoType;
+          let alert = new Alert({
+            text: 'The payment destination (' + knownAccounts[this.destination].name +') requires you to specify a memo to identify your account.',
+            type: Alert.TYPES.WARNING
+          });
+          this.memoAlertGroup.show(alert);
+        }
+
```
Comment (bartekn) [inline controllers/send-widget.controller.es6]: I've tested this. We need to remove the alert when destination is changed to non-exchange address (but the existing memo should be left as it is).

**Comment by ghost** [comment]
Hi, can you take another look? I couldn't just clear out the alerts on destination change event because this would clear out alerts due to other errors as well.

**Comment by bartekn** [comment]
OK, there is still one issue: after sending a transaction to exchange (account in `known_accounts`) the warning box is still visible in the payment form. Other than that, looks fine.

**Comment by ghost** [comment]
OK, hopefully the final version now with an additional improvement and a small bugfix as a bonus.

**Comment by bartekn** [comment]
Thanks!

**Pull request merged**